api/envoy/config/route/v3/route_components.proto

```diff
@@ -403,6 +408,13 @@ message RouteMatch {
     // exactly match the *:path* header once the query string is removed.
     string path = 2;
 
+    // [#not-implemented-hide:]
+    // If this is used as the matcher, the matcher will only match CONNECT requests.
+    // Note that this will not match HTTP/2 upgrade-style CONNECT requests
+    // (WebSocket and the like) as they are normalized in Envoy HTTP/1.1 style
+    // upgrades.
+    ConnectMatcher connect_matcher = 12;
```

Comment:
Arguably this belongs outside the `oneof` because it's categorically different than other path matchers. It's talking about a method rather than path format. It would still be mutually exclusive with path matchers though.